js/path_resolution_test.go

```diff
@@ -0,0 +1,143 @@
+package js
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/lib/testutils"
+)
+
+// This whole file is about tests around https://github.com/grafana/k6/issues/2674
+
+func TestOpenPathResolution(t *testing.T) {
+	t.Parallel()
+	t.Run("simple", func(t *testing.T) {
+		t.Parallel()
+		fs := afero.NewMemMapFs()
+		require.NoError(t, afero.WriteFile(fs, "/path/to/data.txt", []byte("data file"), 0o644))
+		data := `
+		export let data = open("../to/data.txt");
+		if (data != "data file") {
+			throw new Error("wrong content " + data);
+		}
+		export default function() {}
+	`
+		b, err := getSimpleBundle(t, "/path/scripts/script.js", data, fs)
+		require.NoError(t, err)
+
+		_, err = b.Instantiate(testutils.NewLogger(t), 0)
+		require.NoError(t, err)
+	})
+
+	t.Run("intermediate", func(t *testing.T) {
+		t.Parallel()
+		fs := afero.NewMemMapFs()
+		require.NoError(t, afero.WriteFile(fs, "/path/to/data.txt", []byte("data file"), 0o644))
+		require.NoError(t, afero.WriteFile(fs, "/path/another/script/script.js", []byte(`
+            module.exports = open("../../to/data.txt");
+        `), 0o644))
+		data := `
+        let data = require("./../../../another/script/script.js")
```

Comment:
non-blocking nitpick 🙇🏻  I found it hard to process what the test demonstrates with all the `..` symbols, if we could reduce it to the smallest amount that still demonstrates the issue (by for instance simplifying the paths to the files for instance) I would find it beneficial. That's applying to the whole file 👍🏻

Reply:
The whole tests is about how `require` and `open` work with relative paths and what are they *relative* to. 

I did start with somewhat simpler examples but I fast realized that if all my paths are `../otherDir/data.js` it will work always as it files will be on the same fs "depth". So to avoid that I started making them more .... depthly so I can be certain that if I am off by a depth it will give me an error :) 

I might try to get this somewhat nicer when/if we change what either one is relative to.

Reply:
Fair enough, if it does make more sense this way, then it's perfectly fine by me 👍🏻 🙇🏻